### PR TITLE
[VEN-1131] Added methods for repay with bnb

### PR DESCRIPTION
**Pull request by defcon022**
## Description

Resolves #

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/Swap/SwapRouter.sol [hunk 1]
```diff
@@ -315,6 +315,75 @@ contract SwapRouter is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, Rout
...
+    function swapExactTokensForETHAndRepay(
+        address vTokenAddress,
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address[] calldata path,
+        uint256 deadline
+    ) external payable override ensure(deadline) {
+        uint256 balanceBefore = IERC20(path[path.length - 1]).balanceOf(address(this));
+        _swapExactTokensForETH(amountIn, amountOutMin, path, address(this), TypesOfTokens.NON_SUPPORTING_FEE);
+        uint256 balanceAfter = IERC20(path[path.length - 1]).balanceOf(address(this));
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:336]: `_swapExactTokensForETH` is sending BNB (the native token) to `this`, so the check on the balance should not be on the WBNB token, but on the native token

File: contracts/Swap/SwapRouter.sol [hunk 2]
```diff
@@ -315,6 +315,75 @@ contract SwapRouter is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, Rout
...
+        uint256 swapAmount = balanceAfter - balanceBefore;
+        _repay(path[path.length - 1], vTokenAddress, swapAmount);
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:338]: `_repay` is transferring ERC20 tokens, and here we have the native BNB token

File: tests/hardhat/Swap/swapTest.ts [hunk 3]
```diff
@@ -438,5 +451,56 @@ describe("Swap Contract", () => {
           }),
       );
     });
+
+    it("Exact tokens -> BNB", async () => {
+      const deadline = await getValidDeadline();
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      await expect(
```
Comment (chechu) [inline tests/hardhat/Swap/swapTest.ts:458]: What are we expecting here? I would add a test to really check that the debt has been repaid in the market. Not only for this case, but for every case (repay and supply)

File: contracts/Swap/SwapRouter.sol [hunk 4]
```diff
@@ -331,11 +331,14 @@ contract SwapRouter is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, Rout
...
         uint256 deadline
     ) external payable override ensure(deadline) {
-        uint256 balanceBefore = IERC20(path[path.length - 1]).balanceOf(address(this));
+        uint256 balanceBefore = address(this).balance;
         _swapExactTokensForETH(amountIn, amountOutMin, path, address(this), TypesOfTokens.NON_SUPPORTING_FEE);
-        uint256 balanceAfter = IERC20(path[path.length - 1]).balanceOf(address(this));
+        uint256 balanceAfter = address(this).balance;
         uint256 swapAmount = balanceAfter - balanceBefore;
-        _repay(path[path.length - 1], vTokenAddress, swapAmount);
+        uint256 response = IVToken(vTokenAddress).repayBorrowBehalf(msg.sender, swapAmount);
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:338]: For vBNB the method `repayBorrowBehalf` doesn't accept the amount as a parameter. It has to be sent as `msg.value`. See it [here](https://github.com/VenusProtocol/venus-protocol/blob/develop/contracts/Tokens/VTokens/VBNB.sol#L112)

File: contracts/Swap/SwapRouter.sol (hunk 1 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Already fixed.

File: contracts/Swap/SwapRouter.sol (hunk 2 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Already fixed.

File: tests/hardhat/Swap/swapTest.ts (hunk 3 above)
Comment (defcon022) [reply tests/hardhat/Swap/swapTest.ts:458]: I shared same concern with Simeon and Kirill as Simeon was working on it earlier. He said we are just doing UT here and everything else will be covered in the mainnet testing.

File: contracts/Swap/SwapRouter.sol (hunk 4 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Oops, my mistake. FIxed, can verify at [commit](https://github.com/VenusProtocol/venus-protocol/pull/215/commits/7e0d572778b51fbd246174bd16cbf5d1788306f8).

**Review by chechu — APPROVED** [review]

**Review by 0xlucian — APPROVED** [review]

**Comment by chechu** [comment]
LGTM

**Review by 0xlucian — APPROVED** [review]

**Review by narayanprusty — APPROVED** [review]

**Pull request merged**